### PR TITLE
Fix cyberark-pas-account-add (cyberark)

### PR DESCRIPTION
**Pull request by israelpoli**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-28945

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 1]
```diff
@@ -69,6 +69,24 @@ def incident_priority_to_dbot_score(score: float) -> int:
...
+    ordering the properties so that they are valid json for the api
+    """
+    if not properties:
+        return {}
+    if isinstance(properties, dict):
+        return properties
+    elif isinstance(properties, str):
+        try:
+            properties_dict: dict = json.loads(properties.replace("'", '"'))
+            return properties_dict
```
Comment (dorschw) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:83]:
```suggestion
            return json.loads(properties.replace("'", '"'))
```

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 2]
```diff
@@ -69,6 +69,24 @@ def incident_priority_to_dbot_score(score: float) -> int:
...
+        except json.decoder.JSONDecodeError:
+            raise ValueError("Properties are not valid JSON")
```
Comment (dorschw) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:85]:
```suggestion
            raise ValueError("Properties ({properties}) are not valid JSON")
```

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 3]
```diff
@@ -69,6 +69,24 @@ def incident_priority_to_dbot_score(score: float) -> int:
...
+    else:
+        raise ValueError("Properties must be a JSON string or dictionary")
```
Comment (dorschw) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:87]:
```suggestion
    raise ValueError("Properties must be a JSON string or dictionary (got {properties})")
```

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py [hunk 4]
```diff
@@ -361,7 +379,7 @@ def add_account(self,
                     safe_name: str,
                     password: str,
                     secret_type: str,
-                    properties: str,
+                    properties: dict,
```
Comment (dorschw) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:382]:
```suggestion
                    properties: dict | str,
```
right?

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS_test.py [hunk 5]
```diff
@@ -487,3 +487,27 @@ def test_pas_safe_members_list(mocker, response: dict, count_or_total: str, expe
...
+    When:
+        - order_properties_to_dict is called on properties
+    Then:
+        - it returns the expected output
+    """
+    if not isinstance(expected, dict):
+        with pytest.raises(expected):
+            order_properties_to_dict(properties)
+    else:
+        assert order_properties_to_dict(properties) == expected
```
Comment (dorschw) [inline Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS_test.py:513]: Split the test to a positive one (checking successful runs) and a negative (`pytest.raises`)

**Review by dorschw — APPROVED** [review]

File: Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py (hunk 4 above)
Comment (israelpoli) [reply Packs/CyberArkPAS/Integrations/CyberArkPAS/CyberArkPAS.py:382]: It must be a dict, and the `order_properties_to_dict` function forces it to be a dict
(But following the comment I changed the typing for the `add_account_command` function)

**Pull request merged**